CMakeLists.txt

```diff
@@ -1,69 +1,95 @@
+# cmake-format: off
 # CMakeLists.txt -*-CMake-*-
-#
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# cmake-format: on
 
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.27)
 
-project(beman_iter_interface VERSION 0.0.0 LANGUAGES CXX)
+project(
+  beman.iterator_interface
+  VERSION 0.0.0
+  LANGUAGES CXX)
 
+# Local helpers: required to include CompilerFeatureTest.
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
+
+# Includes
+include(CTest)
 include(FetchContent)
 include(CompilerFeatureTest)
 
+# Prechecks.
 beman_iterator_check_deducing_this(COMPILER_SUPPORTS_DEDUCING_THIS)
 
+set(TARGETS_EXPORT_NAME ${CMAKE_PROJECT_NAME}Targets)
+
 option(BEMAN_ITERATOR_INTERFACE_USE_DEDUCING_THIS
-  "Make use of deducing this. Turn this off for non-conforming compilers."
-  ${COMPILER_SUPPORTS_DEDUCING_THIS})
+       "Make use of deducing this. Turn this off for non-conforming compilers."
+       ${COMPILER_SUPPORTS_DEDUCING_THIS})
 
-option(BEMAN_ITERATOR_INTERFACE_ENABLE_TESTING "Build beman.iterator_interface tests" ${PROJECT_IS_TOP_LEVEL})
+option(ITERATOR_INTERFACE_ENABLE_TESTING
+       "Enable building tests and test infrastructure" ${PROJECT_IS_TOP_LEVEL})
 
-if(BEMAN_ITERATOR_INTERFACE_USE_DEDUCING_THIS AND NOT COMPILER_SUPPORTS_DEDUCING_THIS)
-  message(WARNING "Building with deducing this support despite of the compiler's lack of support for it")
+if(BEMAN_ITERATOR_INTERFACE_USE_DEDUCING_THIS
+   AND NOT COMPILER_SUPPORTS_DEDUCING_THIS)
+  message(
+    WARNING
+      "Building with deducing this support despite of the compiler's lack of support for it"
```

Comment:
this is a bit unclear, could you rephrase?

Also end the sentence with a dot

Reply:
Fixed